app/code/Magento/InventoryCatalog/Model/SourceItemsSaveSynchronization/SetDataToLegacyCatalogInventory.php

```diff
@@ -7,140 +7,63 @@
 
 namespace Magento\InventoryCatalog\Model\SourceItemsSaveSynchronization;
 
-use Magento\CatalogInventory\Api\Data\StockItemInterface;
 use Magento\CatalogInventory\Api\StockItemRepositoryInterface;
 use Magento\CatalogInventory\Api\StockItemCriteriaInterfaceFactory;
 use Magento\CatalogInventory\Model\Indexer\Stock\Processor;
 use Magento\CatalogInventory\Model\Spi\StockStateProviderInterface;
-use Magento\CatalogInventory\Model\Stock;
-use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\App\ObjectManager;
+use Magento\InventoryCatalog\Model\LegacySynchronization\Synchronize;
 use Magento\InventoryCatalogApi\Model\GetProductIdsBySkusInterface;
 use Magento\InventoryCatalog\Model\ResourceModel\SetDataToLegacyStockItem;
-use Magento\InventoryCatalogApi\Model\SourceItemsSaveSynchronizationInterface;
 
 /**
  * Set Qty and status for legacy CatalogInventory Stock Item table
+ * @deprecated
```

Comment:
literally saying this class is just part of synchronization plugin, thus I would not keep BC requirement for it and will proceed with modifications 

There is no sense to keep BC for it, because in the code of business logic, you just don't call it at all
```
SetDataToLegacyCatalogInventoryAtSourceItemsSavePlugin

    public function __construct(
        DefaultSourceProviderInterface $defaultSourceProvider,
        IsSourceItemManagementAllowedForProductTypeInterface $isSourceItemsAllowedForProductType,
        GetProductTypesBySkusInterface $getProductTypeBySku,
        SetDataToLegacyCatalogInventory $setDataToLegacyCatalogInventory,
        Synchronize $synchronize = null
    ) {
        $this->defaultSourceProvider = $defaultSourceProvider;
        $this->isSourceItemsAllowedForProductType = $isSourceItemsAllowedForProductType;
        $this->getProductTypeBySku = $getProductTypeBySku;
        $this->synchronize = $synchronize ?:
            ObjectManager::getInstance()->get(Synchronize::class);
    }
```

Reply:
What if someone injected `SetDataToLegacyCatalogInventory` in their business logic? Should not we keep it BC and deprecate it?